lap3dkernel/c++SIMD/lap3dkernel.cpp

```diff
@@ -101,6 +102,45 @@ void test_kernel_vec512(const Real* Xs, const Real* Ys, const Real* Zs,
     }
 }
 
+void test_kernel_vec512_fast_rsqrt(const Real* Xs, const Real* Ys, const Real* Zs,
+                    int ns, const Real* charge,  
+                 const Real* Xt, const Real* Yt, const Real* Zt, int nt, Real* pot){
+#pragma omp parallel for
+    for(int t = 0; t < nt; t += 8){
+
+      Vec8d Xt_vec, Yt_vec, Zt_vec, pot_vec;
+        // load
+        Xt_vec.load(Xt + t);
+        Yt_vec.load(Yt + t);
+        Zt_vec.load(Zt + t);
+        pot_vec.load(pot + t);
+        
+        for(int s = 0; s < ns; s++){
+            Vec8d dx = Vec8d(Xs[s]) - Xt_vec;
+            Vec8d dy = Vec8d(Ys[s]) - Yt_vec;
+            Vec8d dz = Vec8d(Zs[s]) - Zt_vec;
+            Vec8d r2 = dx*dx+dy*dy+dz*dz;
+            
+            // Newton iterations for rsqrt double precision
+            Vec8d rinv = approx_rsqrt(r2);
+            // Note the trick here changes the Newton iteration constants
+            // the standard two Newton iterations are,
+            // rinv *= ((3.0) - r2 * rinv * rinv) * 0.5;
+            // rinv *= ((3.0) - r2 * rinv * rinv) * 0.5;
+            // we can save one simd multiplication " * 0.5 " by using the following trick
+            // first Newton iteration
+            rinv *= ((3.0) - r2 * rinv * rinv);
+            // second Newton iteration
+            rinv *= ((12.0) - r2 * rinv * rinv) * (0.0625);
```

Comment:
The modified Newton iteration saves one simd multiplication, the standard Newton iteration should be 
rinv *= ((3.0) - r2 * rinv * rinv) * 0.5